src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -875,24 +875,27 @@ def forward(
         attention_mask: Optional[torch.Tensor] = None,
         image_hidden_states: Optional[torch.Tensor] = None,
         image_attention_mask: Optional[torch.Tensor] = None,
+        cross_attention_gate: Optional[torch.Tensor] = None,
```

Comment:
is there no default gate function here

Reply:
No because this gate is created from the image_attention_mask which should always be provided, and has a shape dependent on the seq_len.